### PR TITLE
use quote_plus instead of quote

### PR DESCRIPTION
**Pull request by ranchodeluxe**
`asyncpg` doesn't work well for all US-ASCII characters, we need to use `quote_plus` instead

**Related Issue(s):**

https://github.com/MagicStack/asyncpg/pull/1159

**Description:**

See PR and related ticket above

**PR Checklist:**

- [ ] `pre-commit` hooks pass locally
- [ ] Tests pass (run `make test`)
- [ ] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by vincentsarago** [comment]
@ranchodeluxe I'm trying to reproduce the issue using a password like `a2Vw:yk=)CdSis[fek]tW=` (and not applying your fix) but I don't see any thing failing. Are there any other particular things to get errors?

**Comment by ranchodeluxe** [comment]
> a2Vw:yk=)CdSis[fek]tW=

Sorry for the late reply @vincentsarago. I couldn't run tests here a few weeks ago gave up. 

The `/` was the problem in the other repo. Try the password `a2Vw:yk=)CdSis[fek]tW=/o`.

For what it's worth still can't run the test docker image locally. It's failing to pip install things correctly in site-packages for the image user "newuser". Haven't spent much time on it but can try later

**Comment by ranchodeluxe** [comment]
Okay, just approve the workflow please and we should see tests fail 😉

File: stac_fastapi/pgstac/config.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """Postgres API configuration."""
 
 from typing import List, Type
-from urllib.parse import quote
+from urllib.parse import quote_plus as quote
```
Comment (vincentsarago) [inline stac_fastapi/pgstac/config.py:4]: re-added this after confirming locally that the test were failing!

File: tests/conftest.py [hunk 2]
```diff
@@ -53,11 +54,9 @@ def database(postgresql_proc):
         port=postgresql_proc.port,
         dbname="pgstactestdb",
         version=postgresql_proc.version,
-        password="secret",
+        password="a2Vw:yk=)CdSis[fek]tW=/o",
     ) as jan:
-        connection = (
-            f"postgresql://{jan.user}:{jan.password}@{jan.host}:{jan.port}/{jan.dbname}"
-        )
+        connection = f"postgresql://{jan.user}:{quote(jan.password)}@{jan.host}:{jan.port}/{jan.dbname}"
```
Comment (vincentsarago) [inline tests/conftest.py:59]: in the test we also needed to `quote` the password.

**Comment by vincentsarago** [comment]
> For what it's worth still can't run the test docker image locally. It's failing to pip install things correctly in site-packages for the image user "newuser". Haven't spent much time on it but can try later

Yeah sorry about that, I though I fixed this but it doesn't seems to be the case.

for what is worth we've made it easier to just run the test locally (in a virtual env) because now it doesn't touch your local postgres instance. 

Once all the test are ✅ I'll merge and update the changelog.

thanks @ranchodeluxe 🙏

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**